engine/src/main/java/org/terasology/world/propagation/light/InternalLightProcessor.java

```diff
@@ -64,12 +64,14 @@ private static void populateSunlight(LitChunk chunk) {
         PropagationRules sunlightRules = new SunlightPropagationRules(chunk);
         BatchPropagator lightPropagator = new StandardBatchPropagator(sunlightRules, new SingleChunkView(sunlightRules, chunk));
 
+        Vector3i pos = new Vector3i();
         for (int x = 0; x < ChunkConstants.SIZE_X; x++) {
             for (int z = 0; z < ChunkConstants.SIZE_Z; z++) {
                 for (int y = 0; y < ChunkConstants.MAX_SUNLIGHT; ++y) {
-                    Vector3i pos = new Vector3i(x, y, z);
-                    Block block = chunk.getBlock(x, y, z);
-                    byte light = sunlightRules.getFixedValue(block, pos);
+
+                    byte light = sunlightRules.getFixedValue(
```

Comment:
Good idea on extracting the Vector3i from the nested loops. Why did you decide to leave the byte allocation there? Could that be extracted too?

Also, is this part of the code actually looking for a light or is storing some kind of light intensity in this "light" variable? Can you have look? The latter option might call for a renaming of the variable.